lua/entities/gmod_wire_expression2/core/player.lua

```diff
@@ -676,3 +676,119 @@ end
 e2function entity lastDisconnectedPlayer()
 	return lastLeft
 end
+
+----- Deaths+Spawns, Dev: Vurv, 12/7/19 -----
+local DeathAlert = {} -- table of e2s that have runOnDeath(1)
+local RespawnAlert = {}
+local DeathList = { last = {NULL, NULL, NULL, 0} }
+local RespawnList = { last = {NULL,0} }
+
+hook.Add("PlayerDeath","Exp2PlayerDetDead",function(victim,inflictor,attacker)
+	local entry = {n={},ntypes={},s={},stypes={},size=0} -- default table
+	entry.s["Victim"]=victim
+	entry.s["Inflictor"]=inflictor
+	entry.s["Attacker"]=attacker
+	entry.s["Timestamp"]=CurTime()
+	entry.stypes["Victim"]="e"
+	entry.stypes["Inflictor"]="e"
+	entry.stypes["Attacker"]="e"
+	entry.stypes["Timestamp"]="n"
+	entry.size=4
+	DeathList[victim:EntIndex()] = entry -- victim's death is saved in victims death list.
+	DeathList.last = entry -- the most recent death's table is stored here for later use.
+    for ex,_ in pairs(DeathAlert) do -- loops over all chips in deathalert, ignores key.
+        if IsValid(ex) then
+            ex.context.data.runByDeath = entry
+            ex:Execute()
+            ex.context.data.runByDeath = nil
+        end
+    end
+end)
+
+hook.Add("PlayerSpawn","Exp2PlayerDetRespn",function(ply,transition)
+	local entry = {n={},ntypes={},s={},stypes={},size=0} -- default table
```

Comment:
I feel like this table creation shouldn't be duplicated here

Reply:
Having a single default at the top of the file that can be `table.Copy`'d every time would be better. The problem of changing the table structure can't be helped much as it's implied everywhere in E2 that creates tables.